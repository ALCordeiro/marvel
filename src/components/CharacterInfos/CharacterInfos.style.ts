import styled from 'styled-components';
import { IThumbnailImg } from '../../interfaces/characters.interface';

export const CharacterInfosContainer = styled.div`
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
    align-items: flex-start;
    min-width: 100%;
`;

export const CharacterName = styled.h1`
    font-size: 32px;
    padding-bottom: 20px;
`;

export const CharacterDescription = styled.h2`
    font-size: 16px;
    padding-bottom: 40px;
    font-weight: 400;
`;

export const CharacterExtraInfos = styled.div`
    display: flex;
    justify-content: space-between;
    min-width: 200px;
    align-items: center;

    @media (max-width: 800px) {
        align-self: center;
    }
`;

export const CharacterExtraInfosContainer = styled.div`
    display: flex;
    flex-direction: column;
    width: 80px;
`;

export const CharacterExtraInfosText = styled.p`
    font-size: 16px;
    font-weight: bold;
`;

export const CharacterExtraInfosTitle = styled.h3`
    font-size: 16px;
`;

export const CharacterExtraInfosData = styled.div`
    display: flex;
    padding: 20px 0px;
    justify-content: space-between;
    align-items: flex-start;
    font-size: 20px;

    img {
        width: 25px;
        height: 25px;
    }
`;

export const CharacterSubtitle = styled.h4`
    font-size: 18px;
    padding-top: 50px;
    padding-bottom: 30px;
`;

export const CharacterMainInfos = styled.div`
    display: flex;
    min-width: 100%;

    @media (max-width: 800px) {
        flex-direction: column;
        position: relative;
        text-align: center;
        margin-bottom: 20px;
    }
`;

export const CharacterMainTextInfos = styled.div`
    flex-direction: column;
    flex-basis: 70%;
`;

const urlImg = (props: IThumbnailImg) => `
    ${props.thumbnail.path}.${props.thumbnail.extension}`;

export const Avatar = styled.div`
    background: url(${urlImg}) no-repeat center;
    height: 375px;
    max-height: 375px;
    background-size: contain;
    display: block;
    transition: all 1s;
    flex-basis: 30%;

    @media (max-width: 800px) {
        flex-basis: auto;
        width: 100%;
    }
`;
