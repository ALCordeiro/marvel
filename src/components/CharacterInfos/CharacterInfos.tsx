import React from 'react';

import {
    CharacterInfosContainer,
    CharacterName,
    CharacterDescription,
    CharacterExtraInfos,
    CharacterSubtitle,
    CharacterExtraInfosContainer,
    CharacterExtraInfosTitle,
    CharacterExtraInfosData,
    CharacterExtraInfosText,
    CharacterMainInfos,
    CharacterMainTextInfos,
    Avatar,
} from './CharacterInfos.style';
import {
    ICharacterDetail,
    ICharacterMap,
} from '../../interfaces/characters.interface';

export const CharacterInfosComponent: React.FC<ICharacterDetail> = ({
    character,
}: ICharacterDetail) => {
    const dataNotFound = 'Dados não encontrados';
    return (
        <CharacterInfosContainer>
            <CharacterMainInfos>
                <CharacterMainTextInfos>
                    <CharacterName>{character.name}</CharacterName>
                    <CharacterDescription>
                        {character.description
                            ? character.description
                            : dataNotFound}
                    </CharacterDescription>
                </CharacterMainTextInfos>
                <Avatar thumbnail={character.thumbnail} />
            </CharacterMainInfos>

            <CharacterExtraInfos>
                <CharacterExtraInfosContainer>
                    <CharacterExtraInfosTitle>
                        Quadrinhos
                    </CharacterExtraInfosTitle>
                    <CharacterExtraInfosData>
                        <img
                            src={`${window.location.origin}/assets/comics-marvel.svg`}
                            alt="marvel-super-hero"
                        />
                        <CharacterExtraInfosText>
                            {character.comics && character.comics.available
                                ? character.comics.available
                                : dataNotFound}
                        </CharacterExtraInfosText>
                    </CharacterExtraInfosData>
                </CharacterExtraInfosContainer>

                <CharacterExtraInfosContainer>
                    <CharacterExtraInfosTitle>Filmes</CharacterExtraInfosTitle>
                    <CharacterExtraInfosData>
                        <img
                            src={`${window.location.origin}/assets/video-marvel.svg`}
                            alt="marvel-super-hero"
                        />
                        <CharacterExtraInfosText>
                            {character.comics && character.comics.available
                                ? character.comics.available
                                : dataNotFound}
                        </CharacterExtraInfosText>
                    </CharacterExtraInfosData>
                </CharacterExtraInfosContainer>
            </CharacterExtraInfos>
            <CharacterSubtitle>Últimos lançamentos</CharacterSubtitle>
            {character.comics.items.map((characterComics: ICharacterMap) => (
                <p>{characterComics.name}</p>
            ))}
        </CharacterInfosContainer>
    );
};

export default CharacterInfosComponent;
