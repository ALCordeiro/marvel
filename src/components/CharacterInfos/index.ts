export { default as CharacterInfosComponent } from './CharacterInfos';
