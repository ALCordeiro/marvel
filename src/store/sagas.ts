import { all } from 'redux-saga/effects';
import { CharactersSaga } from './ducks/characters/sagas';

export function* rootSagas() {
    yield all([...CharactersSaga]);
}
