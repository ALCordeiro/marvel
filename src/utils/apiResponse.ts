export const characters: any = [
    {
        comics: {
            available: 12,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/comics',
            items: Array(12),
            returned: 12,
        },
        description: '',
        events: {
            available: 1,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/events',
            items: Array(1),
            returned: 1,
        },
        id: 1011334,
        modified: '2014-04-29T14:18:17-0400',
        name: '3-D Man',
        resourceURI: 'http://gateway.marvel.com/v1/public/characters/1011334',
        series: {
            available: 3,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/series',
            items: Array(3),
            returned: 3,
        },
        stories: {
            available: 21,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/stories',
            items: Array(20),
            returned: 20,
        },
        thumbnail: {
            path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
            extension: 'jpg',
        },
        map: () => 'test',
        length: () => 1,
        filter: () => 'test',
    },
];

export const character: any = {
    state: {
        id: 10,
        name: 'Rubick',
        description: 'The Grand magus',
        comics: {
            available: 1,
            items: [
                {
                    name: 'The Granddd magus',
                },
            ],
        },
        series: {
            available: 1,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1017100/series',
            items: [
                {
                    name: 'FREE COMIC BOOK DAY 2013 1 (2013)',
                    resourceURI:
                        'http://gateway.marvel.com/v1/public/series/17765',
                },
            ],
            returned: 1,
        },
        thumbnail: {
            extension: 'jpg',
            path: 'http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16',
        },
    },
};
