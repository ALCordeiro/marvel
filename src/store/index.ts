import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { CharactersState } from './ducks/characters/types';
import rootReducer from './ducks/rootReducer';
import { rootSagas } from './sagas';

export interface ApplicationState {
    characters: CharactersState;
}

const middleware = [];
const initialState = {};
const sagaMiddleware = createSagaMiddleware();
middleware.push(sagaMiddleware);

const store = createStore(
    rootReducer,
    initialState,
    composeWithDevTools(applyMiddleware(...middleware)),
);

sagaMiddleware.run(rootSagas);

export default store;
