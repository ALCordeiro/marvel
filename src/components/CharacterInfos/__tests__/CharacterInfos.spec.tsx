/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { CharacterInfosComponent } from '../CharacterInfos';
import store from '../../../store';
import { character } from '../../../utils/apiResponse';

let wrapper;

beforeEach(() => {
    wrapper = render(
        <Provider store={store}>
            <Router>
                <CharacterInfosComponent character={character.state} />
            </Router>
        </Provider>,
    );
});

test('should render', async () => {
    const { container } = wrapper;
    expect(container.firstChild);
});

test('should match snapshot', () => {
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
});

test('should render with a div as first child', () => {
    const { container } = wrapper;
    expect(container.firstChild.nodeName).toBe('DIV');
});

test('should show the correct text for title', () => {
    expect(screen.getByRole('heading', { level: 1 }).textContent).toBe(
        'Rubick',
    );
});

test('should show the correct text for description', () => {
    expect(screen.getByRole('heading', { level: 2 }).textContent).toBe(
        'The Grand magus',
    );
});
