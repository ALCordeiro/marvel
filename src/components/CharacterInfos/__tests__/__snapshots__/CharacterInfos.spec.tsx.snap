// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="sc-bdnxRM ccnJNr"
>
  <div
    class="sc-pNWdM hyWtTG"
  >
    <div
      class="sc-jrsJWt QVpcv"
    >
      <h1
        class="sc-gtsrHT djwUAP"
      >
        Rubick
      </h1>
      <h2
        class="sc-dlnjwi hJOoTO"
      >
        The Grand magus
      </h2>
    </div>
    <div
      class="sc-kEqXSa bdRqqF"
    />
  </div>
  <div
    class="sc-hKFxyN fXgtiC"
  >
    <div
      class="sc-eCApnc bYkoON"
    >
      <h3
        class="sc-gKAaRy bBfxiC"
      >
        Quadrinhos
      </h3>
      <div
        class="sc-iCoGMd gPvHje"
      >
        <img
          alt="marvel-super-hero"
          src="http://localhost/assets/comics-marvel.svg"
        />
        <p
          class="sc-jSFjdj gZHEXm"
        >
          1
        </p>
      </div>
    </div>
    <div
      class="sc-eCApnc bYkoON"
    >
      <h3
        class="sc-gKAaRy bBfxiC"
      >
        Filmes
      </h3>
      <div
        class="sc-iCoGMd gPvHje"
      >
        <img
          alt="marvel-super-hero"
          src="http://localhost/assets/video-marvel.svg"
        />
        <p
          class="sc-jSFjdj gZHEXm"
        >
          1
        </p>
      </div>
    </div>
  </div>
  <h4
    class="sc-fujyAs hfORVz"
  >
    Últimos lançamentos
  </h4>
  <p>
    The Granddd magus
  </p>
</div>
`;
